crates/bevy_image/src/image.rs

```diff
@@ -71,98 +78,341 @@ macro_rules! feature_gate {
+            ImageFormat::Ktx2 => &["ktx2"],
+            #[cfg(feature = "pnm")]
+            ImageFormat::Pnm => &["pam", "pbm", "pgm", "ppm"],
+            #[cfg(feature = "png")]
+            ImageFormat::Png => &["png"],
+            #[cfg(feature = "qoi")]
+            ImageFormat::Qoi => &["qoi"],
+            #[cfg(feature = "tga")]
+            ImageFormat::Tga => &["tga"],
+            #[cfg(feature = "tiff")]
+            ImageFormat::Tiff => &["tif", "tiff"],
+            #[cfg(feature = "webp")]
+            ImageFormat::WebP => &["webp"],
+        }
+    }
+
+    /// Gets the MIME types for a given format.
+    ///
+    /// If a format doesn't have any dedicated MIME types, this list will be empty.
+    pub const fn to_mime_types(&self) -> &'static [&'static str] {
+        match self {
+            #[cfg(feature = "avif")]
+            ImageFormat::Avif => &["image/avif"],
+            #[cfg(feature = "basis-universal")]
+            ImageFormat::Basis => &["image/basis", "image/x-basis"],
+            #[cfg(feature = "bmp")]
+            ImageFormat::Bmp => &["image/bmp", "image/x-bmp"],
+            #[cfg(feature = "dds")]
+            ImageFormat::Dds => &["image/vnd-ms.dds"],
+            #[cfg(feature = "hdr")]
+            ImageFormat::Hdr => &["image/vnd.radiance"],
+            #[cfg(feature = "gif")]
+            ImageFormat::Gif => &["image/gif"],
+            #[cfg(feature = "ff")]
+            ImageFormat::Farbfeld => &[],
+            #[cfg(feature = "ico")]
+            ImageFormat::Ico => &["image/x-icon"],
+            #[cfg(feature = "jpeg")]
+            ImageFormat::Jpeg => &["image/jpeg"],
+            #[cfg(feature = "ktx2")]
+            ImageFormat::Ktx2 => &["image/ktx2"],
+            #[cfg(feature = "png")]
+            ImageFormat::Png => &["image/png"],
+            #[cfg(feature = "qoi")]
+            ImageFormat::Qoi => &["image/qoi", "image/x-qoi"],
+            #[cfg(feature = "exr")]
+            ImageFormat::OpenExr => &["image/x-exr"],
+            #[cfg(feature = "pnm")]
+            ImageFormat::Pnm => &[
+                "image/x-portable-bitmap",
+                "image/x-portable-graymap",
+                "image/x-portable-pixmap",
+                "image/x-portable-anymap",
+            ],
+            #[cfg(feature = "tga")]
+            ImageFormat::Tga => &["image/x-targa", "image/x-tga"],
+            #[cfg(feature = "tiff")]
+            ImageFormat::Tiff => &["image/tiff"],
+            #[cfg(feature = "webp")]
+            ImageFormat::WebP => &["image/webp"],
+        }
+    }
+
     pub fn from_mime_type(mime_type: &str) -> Option<Self> {
         Some(match mime_type.to_ascii_lowercase().as_str() {
-            "image/avif" => ImageFormat::Avif,
+            // note: farbfeld does not have a MIME type
+            "image/avif" => feature_gate!("avif", Avif),
+            "image/basis" | "image/x-basis" => feature_gate!("basis-universal", Basis),
```

Comment:
This was omitted before; I tried looking and it appears that basis-universal is in the same state as QOI, where people kind of assume that a MIME type exists but there isn't actually an official one filed. So, that's why I included both here.